src/binding.cpp

```diff
@@ -61,52 +61,52 @@ void ExtractOptions(Local<Object> options, void* cptr, sass_context_wrapper* ctx
     ctx_w->request.data = ctx_w;
 
     // async (callback) style
-    Local<Function> success_callback = Local<Function>::Cast(options->Get(NanNew("success")));
-    Local<Function> error_callback = Local<Function>::Cast(options->Get(NanNew("error")));
+    v8::Local<v8::Function> success_callback = v8::Local<v8::Function>::Cast(Nan::Get(options, Nan::New("success").ToLocalChecked()).ToLocalChecked());
+    v8::Local<v8::Function> error_callback = v8::Local<v8::Function>::Cast(Nan::Get(options, Nan::New("error").ToLocalChecked()).ToLocalChecked());
 
-    ctx_w->success_callback = new NanCallback(success_callback);
-    ctx_w->error_callback = new NanCallback(error_callback);
+    ctx_w->success_callback = new Nan::Callback(success_callback);
+    ctx_w->error_callback = new Nan::Callback(error_callback);
   }
 
   if (!is_file) {
-    ctx_w->file = create_string(options->Get(NanNew("file")));
+    ctx_w->file = create_string(Nan::Get(options, Nan::New("file").ToLocalChecked()));
     sass_option_set_input_path(sass_options, ctx_w->file);
   }
 
-  int indent_len = options->Get(NanNew("indentWidth"))->Int32Value();
+  int indent_len = Nan::Get(options, Nan::New("indentWidth").ToLocalChecked()).ToLocalChecked()->Int32Value();
```

Comment:
@kkoopa regarding the proper application of `Maybe<>` types:

Can you have a look at https://github.com/saper/node-sass/commit/df65d0aece948f90972d2286717e88b002f41579 - I have restructured the `Color` class construction to use `Maybe<>` types and not use `IsNumber()` or something like that.

Looks like a pretty extreme functional style - is this a proper way to go?